File: .guides/tests/ch-2.js

keyPressedEvent  = null;

$.getScript(window.location.origin + '/public/js/' + window.testEnv.cmd + '.js?_=' + Date.now())
.done(function (script, status) {
  
  createRandomMaze();
    
  var _left = false;
    
  if(typeof keyPressedEvent  == 'function') {
  
    player.moveLeft = function () {
      _left = true;
    }

    keyPressedEvent('LEFT');

    
    if(!_left) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }   
  }
  else {
    return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
  }
  
  codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.SUCCESS, 'Well done!');    

})
.fail(function (jqxhr, settings, exception) {
  console.log(exception);
  codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, exception.message); 
});

File: .guides/tests/ch-2.py

direction = ''

def moveLeft():
  global direction
  direction = 'LEFT'

try:
  execfile('/home/codio/workspace/public/py/ch-2.py')
  
  keyPressedEvent('LEFT')
  if direction != 'LEFT':
    raise ValueError('keyPressedEvent is not handling the LEFT keyCode correctly')
  
  print 'well done'
  exit(0)
  
except (IOError, SyntaxError, ValueError) as e:
  print str(e)
  exit(1)
  
print 'Not quite right, try again!'
exit(1)
